docs/development/component-checklist.md

```diff
@@ -125,6 +125,12 @@ This document provides a checklist for them that you can walk through.
 
    <sub>[*] It is required because if a component deployed in the Shoot cluster does not specify a Seccomp profile and cannot run with the `RuntimeDefault` Seccomp profile, then enabling the `.spec.kubernetes.kubelet.seccompDefault` field in the Shoot spec would break the corresponding component.</sub>
 
+9. **Use strict mode when decoding data** ([example](https://github.com/gardener/gardener/blob/485c25124ea536e4610c627109017dd34d434921/pkg/provider-local/controller/worker/actuator.go#L52))
```

Comment:
I agree that strict decoding should be imposed on admission components. But do we have to impose it for decoders used in controllers? It will fail early and won't reconcile the resource with the unknown field which sounds good from security PoV. But on the other side, the reconciliation will fail endlessly?
I am just thinking out loud. Because in the past I recall agreeing on using the strict decoding for admission but I am not sure if we have to impose it also for controllers.

@gardener/gardener-maintainers WDYT?

Reply:
There might be examples where strict encoding for controllers can be problematic if APIs are changed, .e.g, a new field was added but controllers are running with the previous API definition for a while.